src/sysrepo.c

```diff
@@ -854,12 +854,20 @@ sr_install_module(sr_conn_ctx_t *conn, const char *schema_path, const char *sear
     if ((err_info = sr_shmmain_ly_ctx_init(&tmp_ly_ctx))) {
         goto cleanup;
     }
-    if ((err_info = sr_lydmods_parse(tmp_ly_ctx, &sr_mods))) {
+    /* create second temporary context to free memory correctly in sr_mods */
+    if ((err_info = sr_shmmain_ly_ctx_init(&tmp_ly_ctx2))) {
+        goto cleanup;
+    }
+    /* create a bond with the second temporary context */
+    if ((err_info = sr_lydmods_parse(tmp_ly_ctx2, &sr_mods))) {
         goto cleanup;
     }
+    /* use first temporary context to load modules */
     if ((err_info = sr_lydmods_ctx_load_modules(sr_mods, tmp_ly_ctx, 1, 1, 0, NULL))) {
         goto cleanup;
     }
+    lyd_free_all(sr_mods);
+    ly_ctx_destroy(tmp_ly_ctx2, NULL);
```

Comment:
Always be careful about your cleanup, in this case both these variables will leak for any `cleanup` above them.